src/datumaro/experimental/export_import.py

```diff
@@ -0,0 +1,443 @@
+                if img_data is not None:
+                    try:
+                        # Handle different image formats
+                        if len(img_data.shape) == 2:
+                            # Grayscale mask
+                            pil_img = Image.fromarray(img_data.astype(np.uint8))
+                        elif len(img_data.shape) == 3:
+                            if img_data.shape[2] == 1:
+                                # Single channel mask
+                                pil_img = Image.fromarray(img_data[:, :, 0].astype(np.uint8))
+                            else:
+                                # Multi-channel mask
+                                pil_img = Image.fromarray(img_data.astype(np.uint8))
+                        else:
+                            print(
+                                f"Warning: Unsupported mask shape {img_data.shape} for field {field_name}, idx {idx}"
+                            )
+                            continue
+
+                        # Save as PNG (best for masks - lossless)
+                        rel_path = f"{field_name}_{idx:06d}.png"
+                        abs_path = output_dir / rel_path
+                        pil_img.save(abs_path)
+                        image_paths[field_name][idx] = str(rel_path)
+                    except Exception as e:
+                        print(
+                            f"Warning: Failed to save mask for field {field_name}, idx {idx}: {e}"
+                        )
+                        continue
+
+    return image_paths
+
+
+def export_dataset(
+    dataset: "Dataset[DType]",
+    output_path: Union[str, Path],
+    export_images: bool = True,
+    as_zip: bool = False,
+) -> None:
+    """
+    Export a dataset to disk in a structured format.
+
+    The dataset is exported with the following structure:
+    - data.parquet: The DataFrame in Parquet format
+    - metadata.json: Schema, categories, and image paths
+    - images/: Directory containing exported images (if export_images=True)
+
+    Image format is automatically determined:
+    - ImagePathField: Preserves original format (copied directly)
+    - ImageCallableField: Saved as PNG (lossless)
+    - InstanceMaskCallableField/MaskCallableField: Saved as PNG (best for masks)
+
+    Args:
+        dataset: The dataset to export
+        output_path: Path to export to (directory or .zip file)
+        export_images: Whether to export images from callable/path fields
+        as_zip: Whether to package everything as a ZIP file
+
+    Raises:
+        ValueError: If the dataset has transforms (not yet supported)
+    """
+    output_path = Path(output_path)
+
+    # Check if dataset has transforms
+    if dataset._transforms is not None:
+        raise ValueError(
+            "Exporting datasets with transforms is not yet supported. "
+            "Please materialize the dataset first by converting it."
```

Comment:
Transforms are an implementation detail, correct if I'm wrong. I think a normal use wouldn't understand this error message. How do you materialize a dataset in practice?